tests/app/main/test_frameworks.py

```diff
@@ -4980,8 +4988,35 @@ def test_shows_signer_details_and_no_form_if_already_agreed(self, data_api_clien
         assert res.status_code == 200
         assert len(doc.xpath('//h2[contains(text(), "Contract variation status")]')) == 1
         assert u"<span>William Drăyton<br />agreed@email.com<br />Friday 19 August 2016 at 4:47pm</span>" in page_text
+        assert u"<span>Waiting for CCS to countersign</span>" in page_text
+        assert len(doc.xpath('//label[contains(text(), "I accept these proposed changes")]')) == 0
+        assert len(doc.xpath('//input[@value="I accept"]')) == 0
+
+    def test_shows_signer_details_and_different_text_if_already_agreed_but_no_countersign(self, data_api_client):
+        already_agreed = self.good_supplier_framework.copy()
+        already_agreed['frameworkInterest']['agreedVariations'] = {
+            "1": {
+                "agreedAt": "2016-08-19T15:47:08.116613Z",
+                "agreedUserId": 1,
+                "agreedUserEmail": "agreed@email.com",
+                "agreedUserName": u"William Drăyton",
+            }}
+        data_api_client.get_framework.return_value = self.g9_framework
+        data_api_client.get_supplier_framework_info.return_value = already_agreed
+
+        res = self.client.get("/suppliers/frameworks/g-cloud-9/contract-variation/1")
+        page_text = res.get_data(as_text=True)
+        doc = html.fromstring(page_text)
+
+        assert res.status_code == 200
+        assert len(doc.xpath('//h1[contains(text(), "G-Cloud 9: proposed contract variation")]')) == 1
+        assert len(doc.xpath('//h2[contains(text(), "Contract variation status")]')) == 1
+        assert u"<span>William Drăyton<br />agreed@email.com<br />Friday 19 August 2016 at 4:47pm</span>" in page_text
```

Comment:
We shouldn't need the `u` in front of strings anywhere since we are Python 3 and everything is unicode by default.  Would be good to remove these as we come across them, rather than adding more.